indexer/eventprocessor.go

```diff
@@ -115,13 +115,17 @@ func (e *EventProcessor) Run() {
 			}
 
 			if err := e.db.Transaction(func(dbTx *gorm.DB) error {
-				hash, err := e.processSlots(dbTx, nextSlot, currentHash)
+				hash, slot, err := e.processSlots(
```

Comment:
If I understand it correctly, when a fork happens, e.processSlots() can roll back one slot for each call. How does it work for epoch process in L135 if we are rolling back multiple epochs?

It seems more logical to update finalized chain status only when it is processing slots forward.

Make e.processEpoch() a standalone func as well. Looks like it only requires node client and db.